src/ros_can/src/node/node_ros_can.cpp

```diff
@@ -0,0 +1,330 @@
+
+#include <canlib.h>
+
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <string>
+
+#include "fs_msgs/msg/control_command.hpp"
+// #include <time.hpp>
+
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/string.hpp"
+
+RosCan::RosCan() : Node("node_ros_can") {
+  operationalStatus =
+      this->create_publisher<custom_interfaces::msg::OperationalStatus>("operationalStatus", 10);
+
+  rlRPMPub = this->create_publisher<custom_interfaces::msg::WheelRPM>("rlRPM", 10);
+  rrRPMPub = this->create_publisher<custom_interfaces::msg::WheelRPM>("rrRPM", 10);
+
+  imuYawAccYPub = this->create_publisher<custom_interfaces::msg::ImuData>("imuYawAccY", 10);
+  imuRollAccXPub = this->create_publisher<custom_interfaces::msg::ImuData>("imuRollAccXPub", 10);
+  imuPitchAccZPub = this->create_publisher<custom_interfaces::msg::ImuData>("imuPitchAccZPub", 10);
+
+  steeringAngleCubeM =
+      this->create_publisher<custom_interfaces::msg::SteeringAngle>("steeringAngleCubeM", 10);
+  controlListener = this->create_subscription<fs_msgs::msg::ControlCommand>(
+      "/as_msgs/controls", 10, std::bind(&RosCan::control_callback, this, std::placeholders::_1));
+  emergencyListener = this->create_subscription<std_msgs::msg::String>(
+      "/as_msgs/emergency", 10,
+      std::bind(&RosCan::emergency_callback, this, std::placeholders::_1));  // maybe change type
+  missionFinishedListener = this->create_subscription<std_msgs::msg::String>(
+      "/as_msgs/mission_finished", 10,
+      std::bind(&RosCan::mission_finished_callback, this,
+                std::placeholders::_1));  // maybe change type
+  // busStatus = this->create_subscription<std_msgs::msg::String>("busStatus", 10,
+  // std::bind(&RosCan::busStatus_callback, this, std::placeholders::_1));
+  timer =
+      this->create_wall_timer(std::chrono::microseconds(500), std::bind(&RosCan::canSniffer, this));
+  // initialize the CAN library
+  canInitializeLibrary();
+  // A channel to a CAN circuit is opened. The channel depend on the hardware
+  hnd = canOpenChannel(1, canOPEN_CAN_FD);
+  // Setup CAN parameters for the channel
+  stat = canSetBusParams(hnd, canBITRATE_500K, 0, 0, 0, 0, 0);  // check these values later
+  if (stat != canOK) {
+    RCLCPP_ERROR(this->get_logger(), "Failed to setup CAN parameters");
+  }
+  // There are different types of controllers, this is the default
+  stat = canSetBusOutputControl(hnd, canDRIVER_NORMAL);
+  if (stat != canOK) {
+    RCLCPP_ERROR(this->get_logger(), "Failed to setup CAN controller");
+  }
+  stat = canBusOn(hnd);
+  if (stat != canOK) {
+    RCLCPP_ERROR(this->get_logger(), "Failed to turn on CAN bus");
+  }
+}
+
+void RosCan::control_callback(fs_msgs::msg::ControlCommand::SharedPtr controlCmd) {
+  // Check if the steering value is within the limits
+  if (controlCmd->steering < STEERING_LOWER_LIMIT || controlCmd->steering > STEERING_UPPER_LIMIT) {
+    RCLCPP_ERROR(this->get_logger(), "Steering value out of range");
```

Comment:
Possibly log the steering value (same for throttle) so that, if the error is thrown, we have a little more information and know, for example, if the value was too high or too low and find the source of the error faster